### PR TITLE
CM Vector list initialization

### PR DESCRIPTION
**Pull request by Attractadore**
This adds array-like list initialization to CM vectors.

### CODE REVIEW

**Comment by aus-intel** [comment]
Thanks for creating PR for this.

Now we have revived our lit testing for cm compiler and this means that contributions should be accompanied with lit tests that cover new functionality/fixed problems. To proceed with this PR you will need to add several lit tests.

To configure cmfe with enabled testing you will need to adjust LLVM and clang cmake lines with the following:

1. LLVM: add -DLLVM_INSTALL_UTILS=ON. This will include FileCheck and other utilities required for lit testing in llvm installation.
2. Clang: add -DCLANG_INCLUDE_TESTS=ON to enable test targets and -DLLVM_EXTERNAL_LIT=<path to lit> to specify where lit tool is located. Lit tool can be found in llvm/utils/lit/lit.py or downloaded as a python package via pip install. In latter case you can omit cmake argument if tool is in your PATH.

After doing this there will be a target "check-cmc" in clang build directory. You can use it to run cm lit tests. Tests sources are located inside test/CMFE directory in this repo. For this particular PR there will be needed 3 types of tests:

1. AST check test. Example can be found in "cmlangspec/2_2_2_a_ast.cpp". This kind of test ensure that expected AST is built.
2. Codegen check test. Example, similarly, is in "cmlangspec/2_2_2_a_codegen.cpp". This is LLVM IR check.
3. Error reporting test. Examples can be found in "member_functions" subdirectory. These will check that correct diagnostic is emitted if source is incorrect. Please mark these tests as XFAIL -- since this is still work in progress to discard this marks and enable proper testing, currently this is the only way to write such tests.

Please add at least one test for each category to your PR and we can continue this review.

**Comment by Attractadore** [comment]
I added some tests. I'm not sure where they should be placed, so I put them all in a single folder for now.

File: test/CMFE/vec_init/vec_init_diag.cpp [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+  vector<unsigned short, 4> v1 = {};                    // OK
+  vector<unsigned short, 4> v2 = {1, 2};                // OK
+  vector<unsigned short, 4> v3 = {1, 2, 3, 5};          // OK
+  vector<unsigned short, 4> v4 = {1, 2, 3, 5, 7, 11};   // error: excess elements
+  vector<unsigned short, 4> v5 = {{1}};                 // warning: braces around scalar
+  vector<unsigned short, 4> v6 = {{1}, 2, 3, 5, 7, 11}; // warning: braces around scalar; error: excess elements
+// CHECK: vec_init_diag.cpp:9:47: error: excess elements in CM vector initializer
+// CHECK: vec_init_diag.cpp:10:35: warning: braces around scalar initializer
+// CHECK: vec_init_diag.cpp:11:35: warning: braces around scalar initializer
+// CHECK: vec_init_diag.cpp:11:49: error: excess elements in CM vector initializer
```
Comment (aus-intel) [inline test/CMFE/vec_init/vec_init_diag.cpp:15]: Is it possible to also cover "err_init_conversion_failed" diagnostic? I see that you modified it so I think there should be a case in this test that triggers this error.

**Review by aus-intel — COMMENTED** [review]
Thanks for adding tests! Creating separate directory for them is OK now. There is not really a lot of structure in current state so any solution would be fine. However, I have comment about one error check.

File: test/CMFE/vec_init/vec_init_diag.cpp (hunk 1 above)
Comment (Attractadore) [reply test/CMFE/vec_init/vec_init_diag.cpp:16]: Fixed.

**Comment by aus-intel** [comment]
Closing this PR as these changes have been merged with #63.

**Pull request closed without merging**